include/biosHandler.hpp

```diff
@@ -4,16 +4,81 @@
 
 namespace vpd
 {
+
+/**
+ * @brief Interface class for BIOS handling.
+ *
+ * The class layouts virtual methods required to be implemented by any concrete
+ * class intended to use feature provided via BIOS handler class.
+ */
+class BiosHandlerInterface
+{
+  public:
+    /**
+     * @brief API to back up or restore BIOS attributes.
+     *
+     * Concrete class should implement the API and read the backed up data from
+     * its designated location and take a call if it should be backed up or
+     * restored.
+     */
+    virtual void backUpOrRestoreBiosAttributes() = 0;
+
+    /**
+     * @brief API to be triggered on BIOS attribute change.
+     *
+     * Concrete class should implement the API to capture any change in the BIOS
+     * attribues and in the implementation needs to filter out the attribute it
+     * is interested.
+     *
+     * @param[in] i_msg - The callback message.
+     */
+    virtual void biosAttribsCallback(sdbusplus::message_t& i_msg) = 0;
+};
+
+/**
+ * @brief IBM specifc BIOS handler class.
+ */
+class IbmBiosHandler : public BiosHandlerInterface
```

Comment:
as we have "BiosHandler" class, some one not seeing the header file may guess "IbmBiosHandler" class might be inherited from "BiosHandler" class. 

It might be my misunderstanding, if am wrong please ignore this comment.

Reply:
That someone is required to read the code to understand it and should not assume anything. :)